### PR TITLE
Fix wrong label mapping in batch_inference for label_model

### PR DESCRIPTION
**Pull request by fayejf**
# What does this PR do ?

Fix groundtruth label mapping in `batch_inference` in `label_models.py` and add test. 

**Collection**: ASR

# Changelog 
-  The groundtruth label `batch_inference` was mapped wrongly. Fixed it
-  Add test for `batch_inference`

# Usage
* You can potentially add a usage example below

```python
from nemo.collections.asr.models import EncDecSpeakerLabelModel 
model = EncDecSpeakerLabelModel.from_pretrained('langid_ambernet')
embs, logits, gt_labels, mapped_labels = model.batch_inference(<path_to_manifest>)
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to https://github.com/NVIDIA/NeMo/pull/5278

### CODE REVIEW

**Review by nithinraok — APPROVED** [review]
Thank you

**Comment by nithinraok** [comment]
Jenkins failing, pls fix it and rebase

File: tests/collections/asr/test_speaker_label_models.py [hunk 1]
```diff
@@ -170,3 +173,22 @@ def test_pretrained_ambernet_logits(self, test_data_dir):
...
 
         assert label == "en"
+
+    @pytest.mark.unit
+    def test_pretrained_ambernet_logits_batched(self, test_data_dir):
+        model_name = 'langid_ambernet'
+        lang_model = EncDecSpeakerLabelModel.from_pretrained(model_name)
+        relative_filepath = "an4_speaker/an4/wav/an4_clstk/fash/an255-fash-b.wav"
+        filename = os.path.join(test_data_dir, relative_filepath)
+        device = 'cuda' if torch.cuda.is_available() else 'cpu'
```
Comment (nithinraok) [inline tests/collections/asr/test_speaker_label_models.py:183]: instead of this mark the test cpu or gpu only or both using pytest

**Review by nithinraok — CHANGES_REQUESTED** [review]

**Review by nithinraok — APPROVED** [review]
